engine/utils/String.cpp

```diff
@@ -126,6 +126,7 @@ String::String(char character) : memory()
 String::~String()
 {
     if (OnHeap()) free(str);
+    str = nullptr;
```

Comment:
Fix something?

Reply:
Yeah there were some issues with pointers being freed that weren't allocated. This might be purely an issue caused by the statics, but this helps nonetheless.